### DISCUSSION TITLE
Custom error message for regex constr() validations?

### DISCUSSION THREAD

**Discussion by mecampbellsoup** [pydantic/pydantic#2758] [Question]
For a field like the following:

```python
class CustomerAttributesPyd(PydanticModel):
    email: constr(strip_whitespace=True, min_length=1, regex=EMAIL_REGEX)
```
Is it possible to produce a custom error message? The default message pattern `'string does not match regex "^(?!-)[a-z0-9-]+(?<!-)$"'` is not exactly user-friendly and I'd rather format error messages entirely in my backend API and not in my frontend JS client.

**Comment by PrettyWood** [comment]
Hi @mecampbellsoup 
Currently it's not possible I'm afraid. You'll need your own strict type with custom validation for this
Hope it helps
